src/utils/featheryClient/index.ts

```diff
@@ -552,13 +552,51 @@ export default class FeatheryClient extends IntegrationClient {
     const data = {
       fuser_key: userId,
       extraction_id: extractionId,
-      run_async: runAsync
+      run_async: true
     };
-    return this._fetch(`${AI_URL}ai/vision/`, {
+
+    this._fetch(`${AI_URL}ai/vision/`, {
       headers: { 'Content-Type': 'application/json' },
       method: 'POST',
       body: JSON.stringify(data)
-    }).then((res) => (res ? res.json() : Promise.resolve({})));
+    });
+
+    return new Promise((resolve) => {
+      const CHECK_INTERVAL = 2000;
+      const MAX_TIME = 3 * 60 * 1000;
+      const MAX_ATTEMPTS = MAX_TIME / CHECK_INTERVAL;
+      let attempts = 0;
+
```

Comment:
maybe here we can have 

if (runAsync) return resolve({});

const checkCompletion = ...